### PR TITLE
@feature/conditional loop

### PR DESCRIPTION
**Pull request by ayodeji-ti**

### CODE REVIEW

File: src/components/editor/ComponentPreview.tsx [hunk 1]
```diff
@@ -46,8 +48,9 @@ const importView = (component: any) => {
 }
 
 const ComponentPreview: React.FC<{
-  componentName: string
-}> = ({ componentName, ...forwardedProps }) => {
+  componentName: string,
+  item?: string
```
Comment (vinayak-sachdeva) [inline src/components/editor/ComponentPreview.tsx:52]: can we extend the solution using forwarded props? as we can have nested items as well

File: src/components/inspector/controls/TextControl.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ const TextControl: React.FC<TextControlPropsType> = ({
   placeholder = '',
 }) => {
   const { setValueFromEvent } = useForm()
-  const value = usePropsSelector(name)
+  let value = usePropsSelector(name)
```
Comment (vinayak-sachdeva) [inline src/components/inspector/controls/TextControl.tsx:23]: why?

File: src/components/inspector/panels/components/LoopPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React, { memo } from 'react'
+import SwitchControl from '~components/inspector/controls/SwitchControl'
+import TextControl from '~components/inspector/controls/TextControl'
+
+const LoopPanel = () => {
+    return (<>
+        <SwitchControl label="Looped View" name="loopView" />
+        <TextControl name="list" label="list" />
```
Comment (vinayak-sachdeva) [inline src/components/inspector/panels/components/LoopPanel.tsx:8]: can we create some list control?

**Review by vinayak-sachdeva — CHANGES_REQUESTED** [review]

File: src/components/inspector/panels/components/LoopPanel.tsx (hunk 3 above)
Comment (ayodeji-ti) [reply src/components/inspector/panels/components/LoopPanel.tsx:8]: Okay, I'll look into this

File: src/components/inspector/controls/TextControl.tsx (hunk 2 above)
Comment (ayodeji-ti) [reply src/components/inspector/controls/TextControl.tsx:23]: This was done when I tried JSON stringifying the value, if it was an object, but that didn't work so I'll return it to const

File: src/components/editor/ComponentPreview.tsx (hunk 1 above)
Comment (ayodeji-ti) [reply src/components/editor/ComponentPreview.tsx:52]: I'll experiment with this also

**Pull request closed without merging**